libraries/Microsoft.Bot.Builder.Azure/AzureBlobStorage.cs

```diff
@@ -108,8 +108,18 @@ public AzureBlobStorage(CloudStorageAccount storageAccount, string containerName
 
             await Task.WhenAll(readTasks);
 
-            // Project back the entries that were read, filtering out any entries that were not found
-            return readTasks.Select(readTask => readTask.Result).Where(kvp => kvp.Key != null);
+            // Project back the entries that were read, filtering out any entries that were not found.
+            // This gives us a List<KeyValuePair<>>. 
+            var items = readTasks.Select(readTask => readTask.Result).Where(kvp => kvp.Key != null);
+
+            // Iterate over the KVPs and add them into a dictionary. 

```

Comment:
Since we're already using LINQ here, I would suggest just using `.ToDictionary(kvp => kvp.Key, kvp => kvp.Value)` instead of the noise of hand crafting this yourself.

Reply:
Fixed. Thanks. Had no idea this existed.